pwiz_tools/Skyline/SettingsUI/Irt/CalibrateIrtDlg.cs

```diff
@@ -66,246 +120,537 @@ private void OnLoad(object sender, EventArgs e)
+                var i = 0;
+                foreach (var pep in peps)
+                {
+                    xValues[i] = pep.Irt;
+                    yValues[i] = regressionLine.GetY(pep.Irt);
+                    tooltips[i] = pep.PeptideModSeq;
+                    i++;
+                }
+            }
+            ShowGraph(Resources.CalibrateIrtDlg_btnGraphIrts_Click_Calibrated_iRT_values, xValues, yValues, tooltips,
+                regressionLine, IsRecalibration);
+        }
 
-                if (FixedPointCount < 2 && rowIndex < Items.Count)
-                    Items[rowIndex].FixedPoint = true;
+        private void ShowGraph(string title, double[] xValues, double[] yValues, Dictionary<int, string> tooltips, RegressionLine line, bool xIrt)
+        {
+            var data = new RegressionGraphData
+            {
+                Title = title,
+                LabelX = !xIrt ? Resources.CalibrateIrtDlg_ShowGraph_Measured : Resources.CalibrateIrtDlg_ShowGraph_Old_iRT,
+                LabelY = !xIrt ? Resources.CalibrateIrtDlg_ShowGraph_iRT : Resources.CalibrateIrtDlg_ShowGraph_New_iRT,
+                XValues = xValues,
+                YValues = yValues,
+                Tooltips = tooltips,
+                RegressionLine = line,
+            };
+
+            using (var graph = new GraphRegression(new[] { data }) { Width = 800, Height = 600 })
+            {
+                graph.ShowDialog(this);
+            }
+        }
 
-                return true;
+        private class RegressionOption
+        {
+            public string Name { get; }
+            public List<Tuple<DbIrtPeptide, PeptideDocNode>> MatchedPeptides { get; }
+
+            private RegressionOption(string name, List<Tuple<DbIrtPeptide, PeptideDocNode>> matchedPeptides, RegressionLine regressionLine)
+            {
+                Name = name;
+                MatchedPeptides = matchedPeptides;
+                RegressionLine = regressionLine;
             }
 
-            private void gridView_CurrentCellDirtyStateChanged(object sender, EventArgs e)
+            public bool IsFixedPoint => MatchedPeptides == null || MatchedPeptides.Count == 0;
+
+            public RegressionLine RegressionLine { get; }
+
+            public static IEnumerable<RegressionOption> All(SrmDocument document)
             {
-                if (GridView.IsCurrentCellDirty)
+                yield return new RegressionOption(Resources.RegressionOption_All_Fixed_points, null, null);
+
+                var docPeptides = document.Peptides.Where(nodePep => nodePep.PercentileMeasuredRetentionTime.HasValue && !nodePep.IsDecoy)
+                    .ToDictionary(nodePep => nodePep.ModifiedTarget, nodePep => nodePep); // TODO duplicate peptides?
+
+                var removedValues = new List<Tuple<double, double>>();
+
+                foreach (var standard in IrtStandard.ALL.Where(standard => !ReferenceEquals(standard, IrtStandard.EMPTY)))

```

Comment:
Should this be iterating over Settings.Default.IrtStandardList instead of IrtStandard.ALL so that it includes user-generated standards?
(I don't understand what's going on here, so I might be wrong).

Reply:
I don't know that there would be a use case for this and it might just be confusing for the user, but I could be wrong. Maybe if they have two sets of their own standards and want to calibrate one against the other?

Reply:
I agree with Nick, and I think you have the right use case. Imagine none of the list of standards we know of actually existed. Then the only way to calibrate multiple standards to the same scale would be using Settings.Default.IrtStandardList, because IrtStandard.All would be empty. Since we can't foresee what kind of standards users will create, it is totally valid that they may eventually want to calibrate one custom set with another custom set.

Reply:
Makes sense, I committed a change to iterate over the settings list instead of just the built in ones.